### ISSUE TITLE
yolov11-seg ground truth

### ISSUE THREAD

**Issue by WYL-Projects** [ultralytics/ultralytics#19548]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Is it possible for yolov11-seg to use a mask plot directly as a truth input instead of a polygonal coordinate point set similar to yolo labels.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @WYL-Projects, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users, where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it effectively.

If this is a custom training ❓ Question or feature inquiry, it would be helpful to share further details about your request. For example, if related to custom training, please include information like dataset examples, training logs, and the specific task you are trying to solve. You may also want to review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

To ensure the best support, please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date verified environments, which include all dependencies like [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to assist you promptly. An Ultralytics engineer will review your question and provide further assistance soon. Thank you for your patience! 😊

**Comment by WYL-Projects** [comment]
okay, thank. Can I directly use mask images as truth values during training? Can I convert a polygon point set to a mask without going through it?

**Comment by Y-T-G** [comment]
No

**Comment by WYL-Projects** [comment]
Okay I get it. Thanks for your patience.

**Comment by WYL-Projects** [comment]
Okay I get it. Thanks for your patience.

**Comment by glenn-jocher** [comment]
Yes, you can use mask images directly as ground truth for YOLO11-seg training. The Ultralytics `convert_segment_masks_to_yolo_seg` function converts mask images to YOLO segmentation format automatically by extracting polygon coordinates from masks. This utility handles the conversion without manual polygon annotation. For details, see the Data Converter documentation on converting segmentation masks to YOLO format.

If you need to convert existing polygon annotations to masks programmatically, the `polygon2mask` utility is available in the Data Utils.

**Comment by WYL-Projects** [comment]
@glenn-jocher Okay, thanks. I've taken your advice.

**Comment by glenn-jocher** [comment]
@WYL-Projects you're welcome! If you have any more questions as you continue working with YOLO11-seg or need clarification on mask/polygon conversions, feel free to ask. The team and community are here to help! 🚀
